R/send.R

```diff
@@ -1,32 +1,111 @@
-require(jsonlite)
-require(httr)
-
 #' Send an email with template
 #'
 #' This sends an email with template
 #'
 #' @param api_key Your Stripe API Key
 #'
-#' @param args An optional list which can contain an amount to refund
+#' @param template_name name of template on Mandrill to use
+#' @param variables list of variables used in template to be filled in
+#' @param subject character vector (single element) to populate email subject line
+#' @param recipient character vector of email addresses of recipients
+#' @param sender characer vector of sendes' email addresses
+#' @param contents  contents of email
+#' @param images character vector of PNGs to include in email.  names (minus the suffix)
+#' must match the image declaration in the Mandrill template. Also note that images must 
+#' saved in same directory as calling environment.
+#' @param css whether to include inline CSS or not. 
 #'
 #' @return nothing
 #'
 #' @export
 #'
-mandrill_send_template <- function(api_key, template_name, variables, subject, recipient, sender) {
-  to <- data.frame(email=recipient)
-  merge_vars <- data.frame(rcpt=recipient)
+mandrill_send_template <- function(api_key=NA, 
+                                   template_name=NA, 
+                                   variables=NA, 
+                                   subject=NA, 
+                                   recipient=NA, 
+                                   sender=NA,
+                                   contents=NULL,
+                                   images=NULL,
+                                   css=FALSE) {
+  to <- data.frame(email=recipient, stringsAsFactors = FALSE)
+  merge_vars <- data.frame(rcpt=recipient, stringsAsFactors = FALSE)
   merge_vars$vars <- list(variables)
-  sendData <- list(key=api_key, template_name=template_name, 
-                   template_content=list(), 
-                   message=list(subject=subject, merge_vars=merge_vars, 
-                                to=to, from_email=sender))
+
+  images_out<-NA
+  if(!missing(images)){    
+    process_images <- function(image){
```

Comment:
Function assignments should be moved to a new file: `helperFunctions.R`;
assigning functions inside other functions is an anti-pattern as it reduces modularity, violates dry, and increases run-time. 

on the other hand, this is very clean and well commented code - thank you!